NPE when clicking a message from a deleted input on a stopped node

Version 1.1.1

Possibly related to the NPE on searches that contained results from deleted inputs or stopped nodes that was present in 1.1.RC3?

Message:

Received by
    deleted input on stopped node
Stored in index
    graylog2_{{OPSEC_REDACTION}}
```
2015-06-05 16:14:49,891 [play-akka.actor.default-dispatcher-10] ERROR play - Cannot invoke the action, eventually got an error: java.lang.NullPointerException
2015-06-05 16:14:49,893 [play-akka.actor.default-dispatcher-10] ERROR application -

! @6mcen26cl - Internal server error, for (GET) [/messages/graylog2_{{OPSEC_REDACTION}}/{{OPSEC_REDACTION}}] ->

play.api.Application$$anon$1: Execution exception[[NullPointerException: null]]
        at play.api.Application$class.handleError(Application.scala:296) ~[com.typesafe.play.play_2.10-2.3.9.jar:2.3.9]
        at play.api.DefaultApplication.handleError(Application.scala:402) [com.typesafe.play.play_2.10-2.3.9.jar:2.3.9]
        at play.core.server.netty.PlayDefaultUpstreamHandler$$anonfun$3$$anonfun$applyOrElse$4.apply(PlayDefaultUpstreamHandler.scala:320) [com.typesafe.play.play_2.10-2.3.9.jar:2.3.9]
```

at play.core.server.netty.PlayDefaultUpstreamHandler$$anonfun$3$$anonfun$applyOrElse$4.apply(PlayDefaultUpstreamHandler.scala:320) [com.typesafe.play.play_2.10-2.3.9.jar:2.3.9]
        at scala.Option.map(Option.scala:145) [org.scala-lang.scala-library-2.10.4.jar:na]
        at play.core.server.netty.PlayDefaultUpstreamHandler$$anonfun$3.applyOrElse(PlayDefaultUpstreamHandler.scala:320) [com.typesafe.play.play_2.10-2.3.9.jar:2.3.9]
        at play.core.server.netty.PlayDefaultUpstreamHandler$$anonfun$3.applyOrElse(PlayDefaultUpstreamHandler.scala:316) [com.typesafe.play.play_2.10-2.3.9.jar:2.3.9]
        at scala.concurrent.Future$$anonfun$recoverWith$1.apply(Future.scala:344) [org.scala-lang.scala-library-2.10.4.jar:na]
        at scala.concurrent.Future$$anonfun$recoverWith$1.apply(Future.scala:343) [org.scala-lang.scala-library-2.10.4.jar:na]
        at scala.concurrent.impl.CallbackRunnable.run(Promise.scala:32) [org.scala-lang.scala-library-2.10.4.jar:na]
at play.api.libs.iteratee.Execution$trampoline$.execute(Execution.scala:46) [com.typesafe.play.play-itera
tees_2.10-2.3.9.jar:2.3.9]        at scala.concurrent.impl.CallbackRunnable.executeWithValue(Promise.scala:40) [org.scala-lang.scala-librar
y-2.10.4.jar:na]        at scala.concurrent.impl.Promise$DefaultPromise.tryComplete(Promise.scala:248) [org.scala-lang.scala-libr
ary-2.10.4.jar:na]        at scala.concurrent.Promise$class.complete(Promise.scala:55) [org.scala-lang.scala-library-2.10.4.jar:na]
        at scala.concurrent.impl.Promise$DefaultPromise.complete(Promise.scala:153) [org.scala-lang.scala-library-2.10.4.jar:na]
        at scala.concurrent.Future$$anonfun$map$1.apply(Future.scala:235) [org.scala-lang.scala-library-2.10.4.jar:na]
        at scala.concurrent.Future$$anonfun$map$1.apply(Future.scala:235) [org.scala-lang.scala-library-2.10.4.ja
r:na]
        at scala.concurrent.impl.CallbackRunnable.run(Promise.scala:32) [org.scala-lang.scala-library-2.10.4.jar:
na]
        at akka.dispatch.BatchingExecutor$Batch$$anonfun$run$1.processBatch$1(BatchingExecutor.scala:67) [com.typ
esafe.akka.akka-actor_2.10-2.3.5.jar:na]        at akka.dispatch.BatchingExecutor$Batch$$anonfun$run$1.apply$mcV$sp(BatchingExecutor.scala:82) [com.types
afe.akka.akka-actor_2.10-2.3.5.jar:na]        at akka.dispatch.BatchingExecutor$Batch$$anonfun$run$1.apply(BatchingExecutor.scala:59) [com.typesafe.akk
a.akka-actor_2.10-2.3.5.jar:na]        at akka.dispatch.BatchingExecutor$Batch$$anonfun$run$1.apply(BatchingExecutor.scala:59) [com.typesafe.akk
a.akka-actor_2.10-2.3.5.jar:na]        at scala.concurrent.BlockContext$.withBlockContext(BlockContext.scala:72) [org.scala-lang.scala-library-2
.10.4.jar:na]
        at akka.dispatch.BatchingExecutor$Batch.run(BatchingExecutor.scala:58) [com.typesafe.akka.akka-actor_2.10-2.3.5.jar:na]
        at akka.dispatch.TaskInvocation.run(AbstractDispatcher.scala:41) [com.typesafe.akka.akka-actor_2.10-2.3.5.jar:na]
        at akka.dispatch.ForkJoinExecutorConfigurator$AkkaForkJoinTask.exec(AbstractDispatcher.scala:393) [com.typesafe.akka.akka-actor_2.10-2.3.5.jar:na]        at scala.concurrent.forkjoin.ForkJoinTask.doExec(ForkJoinTask.java:260) [org.scala-lang.scala-library-2.10.4.jar:na]
at scala.concurrent.forkjoin.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1339) [org.scala-lang.scala
-library-2.10.4.jar:na]        at scala.concurrent.forkjoin.ForkJoinPool.runWorker(ForkJoinPool.java:1979) [org.scala-lang.scala-library
-2.10.4.jar:na]        at scala.concurrent.forkjoin.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:107) [org.scala-lang.scal
a-library-2.10.4.jar:na]Caused by: java.lang.NullPointerException: null
        at com.google.common.base.Preconditions.checkNotNull(Preconditions.java:210) ~[com.google.guava.guava-18.0.jar:na]
        at models.descriptions.NodeDescription.<init>(NodeDescription.java:33) ~[graylog-web-interface.graylog-web-interface-1.1.1.jar:1.1.1]
        at controllers.SearchController.showMessage(SearchController.java:511) ~[graylog-web-interface.graylog-web-interface-1.1.1.jar:1.1.1]
        at Routes$$anonfun$routes$1$$anonfun$applyOrElse$9$$anonfun$apply$465.apply(routes_routing.scala:1675) ~[
graylog-web-interface.graylog-web-interface-1.1.1.jar:na]
        at Routes$$anonfun$routes$1$$anonfun$applyOrElse$9$$anonfun$apply$465.apply(routes_routing.scala:1675) ~[
graylog-web-interface.graylog-web-interface-1.1.1.jar:na]
        at play.core.Router$HandlerInvokerFactory$$anon$4.resultCall(Router.scala:264) ~[com.typesafe.play.play_2
.10-2.3.9.jar:2.3.9]        at play.core.Router$HandlerInvokerFactory$JavaActionInvokerFactory$$anon$15$$anon$1.invocation(Router.sca
la:255) ~[com.typesafe.play.play_2.10-2.3.9.jar:2.3.9]        at play.core.j.JavaAction$$anon$1.call(JavaAction.scala:55) ~[com.typesafe.play.play_2.10-2.3.9.jar:2.3.9
]        at play.GlobalSettings$1.call(GlobalSettings.java:67) ~[com.typesafe.play.play_2.10-2.3.9.jar:2.3.9]
        at play.mvc.Security$AuthenticatedAction.call(Security.java:44) ~[com.typesafe.play.play_2.10-2.3.9.jar:2.3.9]
        at play.core.j.JavaAction$$anonfun$11.apply(JavaAction.scala:82) ~[com.typesafe.play.play_2.10-2.3.9.jar:
2.3.9]        at play.core.j.JavaAction$$anonfun$11.apply(JavaAction.scala:82) ~[com.typesafe.play.play_2.10-2.3.9.jar:
2.3.9]        at scala.concurrent.impl.Future$PromiseCompletingRunnable.liftedTree1$1(Future.scala:24) ~[org.scala-lang
.scala-library-2.10.4.jar:na]        at scala.concurrent.impl.Future$PromiseCompletingRunnable.run(Future.scala:24) ~[org.scala-lang.scala-library-2.10.4.jar:na]        at play.core.j.HttpExecutionContext$$anon$2.run(HttpExecutionContext.scala:40) ~[com.typesafe.play.play_2.10-2.3.9.jar:2.3.9]
at play.api.libs.iteratee.Execution$trampoline$.execute(Execution.scala:46) [com.typesafe.play.play-itera
tees_2.10-2.3.9.jar:2.3.9]        at play.core.j.HttpExecutionContext.execute(HttpExecutionContext.scala:32) ~[com.typesafe.play.play_2.10-
2.3.9.jar:2.3.9]        at scala.concurrent.impl.Future$.apply(Future.scala:31) ~[org.scala-lang.scala-library-2.10.4.jar:na]
        at scala.concurrent.Future$.apply(Future.scala:485) ~[org.scala-lang.scala-library-2.10.4.jar:na]        at play.core.j.JavaAction$class.apply(JavaAction.scala:82) ~[com.typesafe.play.play_2.10-2.3.9.jar:2.3.9]
        at play.core.Router$HandlerInvokerFactory$JavaActionInvokerFactory$$anon$15$$anon$1.apply(Router.scala:252) ~[com.typesafe.play.play_2.10-2.3.9.jar:2.3.9]
        at play.api.mvc.Action$$anonfun$apply$1$$anonfun$apply$4$$anonfun$apply$5.apply(Action.scala:130) ~[com.typesafe.play.play_2.10-2.3.9.jar:2.3.9]
        at play.api.mvc.Action$$anonfun$apply$1$$anonfun$apply$4$$anonfun$apply$5.apply(Action.scala:130) ~[com.t
ypesafe.play.play_2.10-2.3.9.jar:2.3.9]
        at play.utils.Threads$.withContextClassLoader(Threads.scala:21) ~[com.typesafe.play.play_2.10-2.3.9.jar:2
.3.9]
        at play.api.mvc.Action$$anonfun$apply$1$$anonfun$apply$4.apply(Action.scala:129) ~[com.typesafe.play.play
_2.10-2.3.9.jar:2.3.9]        at play.api.mvc.Action$$anonfun$apply$1$$anonfun$apply$4.apply(Action.scala:128) ~[com.typesafe.play.play
_2.10-2.3.9.jar:2.3.9]        at scala.Option.map(Option.scala:145) [org.scala-lang.scala-library-2.10.4.jar:na]
        at play.api.mvc.Action$$anonfun$apply$1.apply(Action.scala:128) ~[com.typesafe.play.play_2.10-2.3.9.jar:2.3.9]
        at play.api.mvc.Action$$anonfun$apply$1.apply(Action.scala:121) ~[com.typesafe.play.play_2.10-2.3.9.jar:2.3.9]
        at play.api.libs.iteratee.Iteratee$$anonfun$mapM$1.apply(Iteratee.scala:483) ~[com.typesafe.play.play-ite
ratees_2.10-2.3.9.jar:2.3.9]        at play.api.libs.iteratee.Iteratee$$anonfun$mapM$1.apply(Iteratee.scala:483) ~[com.typesafe.play.play-ite
ratees_2.10-2.3.9.jar:2.3.9]        at play.api.libs.iteratee.Iteratee$$anonfun$flatMapM$1.apply(Iteratee.scala:519) ~[com.typesafe.play.play
-iteratees_2.10-2.3.9.jar:2.3.9]        at play.api.libs.iteratee.Iteratee$$anonfun$flatMapM$1.apply(Iteratee.scala:519) ~[com.typesafe.play.play-iteratees_2.10-2.3.9.jar:2.3.9]        at play.api.libs.iteratee.Iteratee$$anonfun$flatMap$1$$anonfun$apply$14.apply(Iteratee.scala:496) ~[com.typesafe.play.play-iteratees_2.10-2.3.9.jar:2.3.9]
        at play.api.libs.iteratee.Iteratee$$anonfun$flatMap$1$$anonfun$apply$14.apply(Iteratee.scala:496) ~[com.typesafe.play.play-iteratees_2.10-2.3.9.jar:2.3.9]
at scala.concurrent.impl.Future$PromiseCompletingRunnable.liftedTree1$1(Future.scala:24) ~[org.scala-lang
.scala-library-2.10.4.jar:na]
        at scala.concurrent.impl.Future$PromiseCompletingRunnable.run(Future.scala:24) ~[org.scala-lang.scala-library-2.10.4.jar:na]
        ... 6 common frames omitted

--- a/app/controllers/SearchController.java
+++ b/app/controllers/SearchController.java
@@ -508,11 +508,15 @@ public class SearchController extends AuthenticatedController {
             final Input sourceInput = getSourceInput(sourceNode, message);
             final Input sourceRadioInput = getSourceInput(sourceRadio, message);
 
-            nodes.add(new NodeDescription(sourceNode));
+            if (sourceNode != null) {
+                nodes.add(new NodeDescription(sourceNode));
+            }
             if (sourceRadio != null) {
                 nodes.add(new NodeDescription(sourceRadio));
             }
-            inputs.add(new InputDescription(sourceInput));
+            if (sourceInput != null) {
+                inputs.add(new InputDescription(sourceInput));
+            }
             if (sourceRadioInput != null) {
                 inputs.add(new InputDescription(sourceRadioInput));
             }
@@ -560,6 +564,7 @@ public class SearchController extends AuthenticatedController {
         }
     }
 
+    @Nullable
     private Node getSourceNode(MessageResult m) {
         try {
             return nodeService.loadNode(m.getSourceNodeId());
@@ -570,6 +575,7 @@ public class SearchController extends AuthenticatedController {
         return null;
     }
 
+    @Nullable
     private Radio getSourceRadio(MessageResult m) {
         if (m.viaRadio()) {
             try {
@@ -582,6 +588,7 @@ public class SearchController extends AuthenticatedController {
         return null;
     }
 
+    @Nullable
     private static Input getSourceInput(Node node, MessageResult m) {
         if (node != null && isPermitted(RestPermissions.INPUTS_READ, m.getSourceInputId())) {
             try {
@@ -594,6 +601,7 @@ public class SearchController extends AuthenticatedController {
         return null;
     }
 
+    @Nullable
     private static Input getSourceInput(Radio radio, MessageResult m) {
         if (radio != null) {
             try {

